sympy/polys/matrices/domainmatrix.py

```diff
@@ -204,13 +918,53 @@ def lu_solve(self, rhs):
         return self.from_rep(sol)
 
     def charpoly(self):
+        r"""
+        Returns the coefficients of the characteristic polynomial of the DomainMatrix
```

Comment:
It's important in these docs to clarify whether or not something is `Expr` or is a domain element. If it is a domain element then in which domain?